lib/TPP/InterchangeBlockConvToExposeMatmul.cpp

```diff
@@ -0,0 +1,84 @@
+//===- InterchangeBlockConvToExposeMatmul.cpp --------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Passes.h"
+#include "TPP/TransformUtils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+
+#define GEN_PASS_CLASSES
+#include "TPP/Passes.h.inc"
+
+namespace {
+
+// Interchange a blocked convolutions to expose a linalg.matmul.
+struct InterchangeBlockConvToExposeMatmulImpl
+    : OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp genericOp,
+                                PatternRewriter &rewriter) const override {
+    if (!linalgx::utils::isBlockedConvolution(genericOp))
+      return failure();
+
+    // clang-format off
```

Comment:
Is this the only form of blocked convolution that `isBlockedConvolution` detects?

Reply:
yes, all the convolutions are blocked the same way.